### PR TITLE
Add type annotations to django.shortcuts

### PR DESCRIPTION
**Pull request by bartoszgrabski**

### CODE REVIEW

File: django/shortcuts.py [hunk 1]
```diff
@@ -31,7 +39,7 @@ def render(request, template_name, context=None, content_type=None, status=None,
     return HttpResponse(content, content_type, status)
 
 
-def redirect(to, *args, **kwargs):
+def redirect(to: Union[str, Any], *args: Any, **kwargs: Any) -> HttpResponseRedirect:
```
Comment (dmoisset) [inline django/shortcuts.py]: Shouldn't the `to` parameter allow a `Model` instead of `Any` here?

File: django/shortcuts.py [hunk 2]
```diff
@@ -65,11 +73,11 @@ def _get_queryset(klass):
     """
     # If it is a model class or anything else with ._default_manager
     if hasattr(klass, '_default_manager'):
-        return klass._default_manager.all()
+        return klass._default_manager.all()  # type: ignore
     return klass
 
 
-def get_object_or_404(klass, *args, **kwargs):
+def get_object_or_404(klass: Union[Model, Manager, QuerySet], *args: Any, **kwargs: Any) -> Model:
```
Comment (dmoisset) [inline django/shortcuts.py]: I think `klass` shouldn't be a `Model` (which mean a specific instance of a Model) but `Type[Model]` which means a Model class. Same applies to `get_list_or_404`

**Review by dmoisset — CHANGES_REQUESTED** [review]
Thanks for the contribution! I reviewed some details, can you take a look at those?

File: django/shortcuts.py (hunk 1 above)
Comment (bartoszgrabski) [reply django/shortcuts.py]: You're right.

File: django/shortcuts.py (hunk 2 above)
Comment (bartoszgrabski) [reply django/shortcuts.py]: Fixed.

**Comment by bartoszgrabski** [comment]
Also updated `_get_queryset` based on current annotations.

**Pull request closed without merging**